components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3276,7 +3277,9 @@ private FilterData getFilterDataForDBQuery(String filter) {
                 Node rootNode = filterTreeBuilder.buildTree();
                 filterData = getFilterDataFromFilterTree(rootNode);
             } catch (IOException | IdentityException e) {
-                log.error("Error occurred while converting filter query with tree builder.", e);
+                log.error("Error occurred while converting filter query with filter tree builder.", e);
```

Comment:
don't log the error and throw it, Then the error log is printed twice.
Can we say this as a client expectation? what could be the reasons for IOException | IdentityException ?

Reply:
The reason for those exception is invalid values in the filter query (filter attribute or filter condition. Will the following error imply it clearly?
`Filter attribute or filter condition is empty or invalid.`